### PR TITLE
ElSwitch: add loading function

### PR DESCRIPTION
**Pull request by ohhoney1**
Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

File: test/unit/specs/switch.spec.js [hunk 1]
```diff
@@ -127,6 +127,29 @@ describe('Switch', () => {
...
+      data() {
+        return {
+          value: false
+        };
+      }
+    }, true);
+
+    vm.$el.querySelector('.el-switch__core').click();
+    expect(vm.$el.querySelector('.el-switch__core .circular')).to.exist;
+    Vue.nextTick(() => {
```
Comment (wacky6) [inline test/unit/specs/switch.spec.js]: We usually use `vm.$nextTick` here.

**Comment by wacky6** [comment]
Thanks for contributing. 

We may have to talk with UI designers about the loading design. Element has a Loading component, it's better to mimic its style. Button's loading design might be of interest.

Interesting feature to have, in case that server responds slowly, the loading flag prevents user from repeatedly clicking the switch.

File: test/unit/specs/switch.spec.js (hunk 1 above)
Comment (ohhoney1) [reply test/unit/specs/switch.spec.js]: I see, but i refer to the file `switch.spec.js`, maybe we need optimization here

https://github.com/ElemeFE/element/blob/dev/test/unit/specs/switch.spec.js#L124-L127
```
Vue.nextTick(() => {
    expect(vm.value).to.true;
    done();
});
```

Comment (wacky6) [reply test/unit/specs/switch.spec.js]: Interesting, I think we will normalize `$nextTick` usage across test codes in next major release.

**Comment by wacky6** [comment]
@jikkai 让交互看一下这个 loading 的设计吧 / Please ask UED to evaluate this loading design.

**Comment by jikkai** [comment]
@ohhoney1 

Thanks for contributing!

According to the designer's judgment, the processing status is reasonable. Unfortunately, they have provided a different UI from this case which means we have to close this pr. But don't worry about that, we will implement it soon :)

**Comment by ohhoney1** [comment]
@jikkai I am looking forward to it.

**Pull request closed without merging**